web_infrastructure/gfdeploy

```diff
@@ -0,0 +1,193 @@
+    user:
+        description:
+            - Administrative user of Glassfish
+        required: true
+        default: null
+    password:
+        description:
+            - Password for accessing the Glassfish administrative console (i.e. password for C(user))
+        required: true
+        default: null
+    force:
+        description:
+            - Forces Glassfish to deploy application even if one with the same name is already deployed.
+        required: false
+        default: false
+    state:
+        description:
+	    - Describes whether to deploy or undeploy
+        choices: ["present", "absent"]
+        default: present
+
+requirements: [ requests, requests-toolbelt ]
+author:
+    - "Nicholas McKinney (@nickmk)"
+"""
+
+EXAMPLES = '''
+# Deploy a web-application named Foo 
+- gfdeploy: host='my-glassfish-server' user=foo password=bar force=true contextroot=Foo name=Foo filepath=/baz/bar/foo.war
+
+# Undeploy a web-application named Foo
+- gfdeploy: host='my-glassfish-server' user=foo password=bar state=absent name=Foo
+'''
+
+RETURN = '''
+changed:
+	description: Boolean representing a change of state in Glassfish due to the module
+        type: boolean
+location:
+	description: URL of the Glassfish server's administrative console used. The application name is suffixed in the case of undeployment.
+        type: string
+        sample: [ "https://my-glassfish:4848/management/domain/applications/application/Foo", "https://my-glassfish:4848/management/domain/applications/application" ]
+response_code:
+	description: HTTP response code Glassfish returned for the operation. 200 equates to success. A 404 during undeployment suggests the application was already absent
+	type: int
+	sample: [ 404, 200 ]
+'''
+
+def getFilename(filepath):
+    head, tail = ntpath.split(filepath)
+    if tail == '':
+        f = ntpath.basename(head)
+    else:
+        f = tail
+    last = f.rfind('.')
+    if last != -1:
+        f = f[:last]
+    return f
+
+def deploy(url, contextroot, name, filepath, force, user, password):
+    params = {'contextroot' : contextroot, 'name' : name, 'force':str(force)}
+    requests.packages.urllib3.disable_warnings()
+    f = MultipartEncoder(
+        fields={'contextroot':contextroot, 'name':name, 'force':str(force), 'id':(getFilename(filepath), open(filepath, 'rb'), 'application/octet-stream')}
+    )
+    r = requests.post(url, verify=False, data=f, headers={'Content-Type':f.content_type, 'X-Requested-By':'ansible'}, auth=(user,password))    
+    if r.status_code == 200:
+        return True, r.status_code
+    return False, r.status_code
+
+def undeploy(url, user, password):
+    requests.packages.urllib3.disable_warnings()
+    r = requests.delete(url, headers={'X-Requested-By':'ansible'}, verify=False, auth=(user,password))
+    return (True, r.status_code) if r.status_code == 200 else (False, r.status_code)
+
+def protocol(secure):
+    return 'https://' if bool(secure) == True else 'http://'
```

Comment:
This ternary syntax is not python2.4 compatible.  Modules should be 2.4 compatible if possible, and by switching to `fetch_url` it looks like this module can be.